src/identity/session.guard.ts

```diff
@@ -20,14 +23,24 @@ export class SessionGuard implements CanActivate {
   constructor(
     private readonly logger: ConsoleLoggerService,
     private userService: UsersService,
+    @Inject(noteConfiguration.KEY)
+    private noteConfig: NoteConfig,
   ) {
     this.logger.setContext(SessionGuard.name);
   }
 
   async canActivate(context: ExecutionContext): Promise<boolean> {
-    const request: Request & { session?: { user: string }; user?: User } =
-      context.switchToHttp().getRequest();
+    const request: Request & {
+      session?: { user: string; authProvider: string };
+      user?: User;
+    } = context.switchToHttp().getRequest();
     if (!request.session?.user) {
+      if (this.noteConfig.guestAccess !== GuestAccess.DENY) {
```

Comment:
I feel like this if-statement (and with it this class) is now complicated enough to warrant some docs/comments.